### PR TITLE
Migrate from Byte to ByteBuffer for req/response

### PR DESCRIPTION
**Pull request by Sammers21**
* migrate from byte to ByteBuffer for req/response
* remove the tk package
* move RequestLine to rq package

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#30](https://github.com/artipie/http/issues/30) is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/82e63fe2-f1aa-4c80-a4da-7fb5c8b79c62); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by Sammers21** [comment]
@iakunin, the conflict has been fixed

File: src/main/java/com/artipie/http/Response.java [hunk 1]
```diff
@@ -36,7 +37,14 @@
     /**
      * Empty response.
      */
-    Response EMPTY = con -> con.accept(0, Collections.emptyList(), new PubEmpty<Byte>());
+    Response EMPTY = con -> {
+        final int code = 200;
```
Comment (iakunin) [inline src/main/java/com/artipie/http/Response.java]: @Sammers21 maybe we can inline this variable? It's pretty obvious what it means without any additional description.

Comment (Sammers21) [reply src/main/java/com/artipie/http/Response.java]: No, CheckStyle would complain

Comment (Guseyn) [reply src/main/java/com/artipie/http/Response.java]: @iakunin why is it so obvious? i would even change var name to the `responseCode`, but stupid check style wouldn't allow it

**Comment by Sammers21** [comment]
@iakunin, review the pull request please

**Comment by g4s8** [comment]
@rultor release, tag=`0.2`

**Comment by 0crat** [comment]
Job audit: performer @iakunin[/z](https://www.0crat.com/u/iakunin)[/z](https://www.0crat.com/u/iakunin) didn't make CR comments

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @iakunin[/z](https://www.0crat.com/u/iakunin)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**